server/game/cards/01_SOR/units/LieutenantChildsenDeathStarPrisonWarden.ts

```diff
@@ -0,0 +1,35 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Aspect, RelativePlayer, Location, TargetMode } from '../../../core/Constants';
+
+export default class LieutenantChildsenDeathStarPrisonWarden extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '2855740390',
+            internalName: 'lieutenant-childsen#death-star-prison-warden'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'Reveal up to 4 Vigilance cards from your hand',
+            targetResolver: {
+                mode: TargetMode.UpTo,
+                numCards: 4,
+                controller: RelativePlayer.Self,
+                locationFilter: Location.Hand,
+                cardCondition: (card) => card.hasSomeAspect(Aspect.Vigilance),
+                immediateEffect: AbilityHelper.immediateEffects.reveal(),
+            },
+            then: (context) => ({
```

Comment:
nit: by convention we call this `thenContext` to differentiate from any other context parameters used inside